File: client/src/game/conf/PropertyTypeConfig.ts
class PropertyTypeConfig {
    private static _dic:Dictionary<PropertyTypeConfig> = new Dictionary<PropertyTypeConfig>();
    private static _arr:Array<PropertyTypeConfig> = new Array<PropertyTypeConfig>();

	 public Id :number ;
     public Parent :number ;
     public Name :string ;
     public Bg :number ;
     public Icon :string ;
     public IconDown :string ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<PropertyTypeConfig>{
        return  PropertyTypeConfig._dic;
    }
    public static get Arr():Array<PropertyTypeConfig>{
        return PropertyTypeConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:PropertyTypeConfig = <PropertyTypeConfig>json[i];
				PropertyTypeConfig._arr.push(config);
				PropertyTypeConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("PropertyTypeConfig loader fail!");
        }
    }
}

File: server/src/game/conf/PropertyTypeConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type PropertyTypeConfigElem struct {
	Id int
	Parent int
	Name string
	Bg int
	Icon string
	IconDown string
	
}

type PropertyTypeConfig0 struct {
	Dic map[int]*PropertyTypeConfigElem
	Arr []*PropertyTypeConfigElem
}

var (
	PropertyTypeConfig PropertyTypeConfig0
)

func loadPropertyTypeConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	PropertyTypeConfig.Dic = make(map[int]*PropertyTypeConfigElem)
	PropertyTypeConfig.Arr = make([]*PropertyTypeConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &PropertyTypeConfig.Arr)
	for _,conf := range PropertyTypeConfig.Arr{
		PropertyTypeConfig.Dic[conf.Id] = conf
	}
	
}
